Makefile

```diff
@@ -15,11 +15,26 @@ else
 	endif
 endif
 
+##@ Build Dependencies
+
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/.bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+## Tool Binaries
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+ENVTEST ?= $(LOCALBIN)/setup-envtest
+
+## Tool Versions
```

Comment:
Could we, for internal consistency use a pattern we have developed at ory? Please take a look [here](https://github.com/ory/k8s/blob/master/Makefile#L52) :) 
The benefit is that we can then refer to the `.deps` directory to easily determine if deps have changed version and rebuild cache if yes. 
This is of course provided we will manage to migrate this repo to GHA 😅

Reply:
I'll leave that one to the Ory folks, it may be a little tricky as some of the tools involved are downloaded via installer scripts and such (this is the default from the kubebuilder v3 makefile).Will require a bit of hacking around.